packages/contracts-bedrock/src/L1/SystemConfigInterop.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Constants } from "src/libraries/Constants.sol";
+import { OptimismPortalInterop as OptimismPortal } from "src/L1/OptimismPortalInterop.sol";
+import { GasPayingToken } from "src/libraries/GasPayingToken.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { ConfigType } from "src/L2/L1BlockInterop.sol";
+
+/// @notice Error thrown when an address has no code.
+error NoCode(address);
+
+/// @title SystemConfigInterop
+/// @notice The SystemConfig contract is used to manage configuration of an Optimism network.
+///         All configuration is stored on L1 and picked up by L2 as part of the derviation of
+///         the L2 chain.
+contract SystemConfigInterop is SystemConfig {
+    /// @notice Modifier to check if an address has code.
+    /// @param _address Address to check.
+    /// @param _check Boolean to determine if the check should be performed.
+    modifier mustHaveCode(address _address, bool _check) {
+        if (_check) {
+            assembly {
+                let size := extcodesize(_address)
+                // Revert if the code size is 0.
+                if iszero(size) {
+                    mstore(0x00, 0x674520b6) // 0x674520b6 is the 4-byte selector of "NoCode(address)"
```

Comment:
Where is the abi encoded address along with the 4byte selector? right now its just returning 4 bytes